homeassistant/components/binary_sensor/bayesian.py

```diff
@@ -92,51 +133,59 @@ def async_threshold_sensor_state_listener(entity, old_state,
             self.watchers[platform](entity_obs)
 
             prior = self.prior
+            print(self.current_obs.values())
             for obs in self.current_obs.values():
-                prior = self._update_probability(obs, prior)
+                prior = update_probability(prior, obs['prob_true'],
+                                           obs['prob_false'])
 
             self.probability = prior
 
-            hass.async_add_job(self.async_update_ha_state, True)
+            self.hass.async_add_job(self.async_update_ha_state, True)
 
         for obs in self._observations:
             entity_id = obs['entity_id']
-            async_track_state_change(hass, entity_id,
+            async_track_state_change(self.hass, entity_id,
                                      async_threshold_sensor_state_listener)
 
     def _process_numeric_state(self, entity_observation):
         entity = entity_observation['entity_id']
-        if condition.async_numeric_state(self._hass, entity,
-                                         entity_observation.get('below'),
-                                         entity_observation.get('above'), None,
-                                         entity_observation):
 
-            self.current_obs[entity] = entity_observation['probability']
+        should_trigger = condition.async_numeric_state(
+            self.hass, entity,
+            entity_observation.get('below'),
+            entity_observation.get('above'), None, entity_observation)
+
+        if should_trigger:
```

Comment:
The code between the `if` and `else` here makes a pattern that is repeated for both observation platforms. Maybe break it out into a helper method?